src/module.c

```diff
@@ -10821,9 +10829,14 @@ void processModuleLoadingProgressEvent(int is_aof) {
     }
 }
 
-/* When a module key is deleted (in dbAsyncDelete/dbSyncDelete/dbOverwrite), it 
+/* When a key is deleted (in dbAsyncDelete/dbSyncDelete/dbOverwrite), it 
 *  will be called to tell the module which key is about to be released. */
 void moduleNotifyKeyUnlink(robj *key, robj *val, int dbid) {
+    server.lazy_expire_disabled++;
```

Comment:
why do we turn off lazy expire here but not before triggering a module KSN callback? what's the difference?

Reply:
or even before calling unlink/unlink2 just a few lines below?

Reply:
It's same with triggering a module KSN callback as this event is KSN callback.
But I think we should turn on lazy expire after unlink/unlink2?

Reply:
i'm not certain i understand the discussion here.
are we talking about other KSNs?
the reason we disabled it for this one is that we expect modules to do RM_OpenKey.
in unlink/unlik2 we don't expect or promote it (also, there's no need for it since you have the value)

Reply:
@MeirShpilraien do you think we should set that flag for all KSN, and also move the decrement to after calling the unlink CB below?
maybe we should.. 
even if we don't document that users can do RM_Open, maybe it's better making this safe (also affects read-only RM_Call in case someone does it in KSN)

Reply:
I think that if we decide to do it here we should also do it for all KSN. We should just consider if it can be a breaking change (I believe not).

Reply:
i don't see why this should be a breaking change, RM_OpenKey and RM_Call will still return the right response, just that the key isn't deleted.
the only breaking possibility i can see is if someone is doing any write action from the KSN, in which case lookupKey will return NULL, the caller will think there's no key and try to create one, and that can crash since the key already exists.
but people should not do write operations from KSN, so i think this is safe.

@huangzhw can you please extend the effect of this flag to also cover the `unlink`/`unlink2` callback and keyspace notification callback, and mention this change in the top comment.

Reply:
done.